### DISCUSSION TITLE
When choosing two topics to merge (if they are similar), how does BERTopic aggregate them?

### DISCUSSION THREAD

**Discussion by tawfiqam** [MaartenGr/BERTopic#709] [General]
I am considering merging topics that are similar. I am wondering how BERTopic aggregates the topic probabilities?

**Comment by MaartenGr** [comment]
It depends on the type of probabilities you have selected. If you have set `calculate_probabilities=False`, then flat topic probabilities are generated. These probabilities indicate the probability of a document belonging to its **predicted** topic. When merging topics, these probabilities are not changed. 

If you have set `calculate_probabilities=True`, then a topic-document probability matrix is generated instead of flat probabilities. In this matrix, you can see the probability of a document belonging to **any** topic. When merging topics, these probabilities are then added to one another since we have information about the topics they are merged together with. 

EDIT: Do note that there is a [new version](https://github.com/MaartenGr/BERTopic/pull/668) coming up that will fix some of the issues when merging topics.

**Reply by tawfiqam** [reply]
Fantastic!  Thank you very much.
